activerecord-virtual_attributes.gemspec

```
@@ -25,7 +25,7 @@ Gem::Specification.new do |spec|
 
   spec.require_paths = ["lib"]
 
-  spec.add_runtime_dependency "activerecord", "~> 6.1.0"
+  spec.add_runtime_dependency "activerecord", "~> #{ENV.fetch("RAILS_VERSION", "6.1")}.0"
```

I'd prefer the gemspec to be hardcoded.

Even if it is a `"> 6.1"` and a specific version is coded into the `Gemfile`

---

I don't think we can do that, because the gemspec overrides it, I think ... I'll have to see

---

Yeah, you can't do that.  If the gemspec hardcodes it you can't change it later unless we "hack" it the way override_gem does it, but that feels worse.

`"> 6.1"` is bad because then when we try to install it outside of this application, it will force Rails 7 when want it to be Rails 6.

Would this be more acceptable?

```suggestion
  if ENV["RAILS_VERSION"]
    spec.add_runtime_dependency "activerecord", "~> #{ENV["RAILS_VERSION"]}.0"
  else
    spec.add_runtime_dependency "activerecord", "~> 6.1.0"
  end
```

---

Actually override_gem style might be preferable - I'll investigate that.

---

The gemspec needs to be static.

---

Yes, I agree - I need to find a way around this, and I want to leave the gemspec alone.